src/helpers/RainbowContext.tsx

```diff
@@ -62,11 +86,55 @@ export default function RainbowContextWrapper({ children }: any) {
 
   const { isDarkMode, setTheme, colors } = useTheme();
 
+  const dispatch = useDispatch();
+  const updateAssetOnchainBalanceIfNeeded = useUpdateAssetOnchainBalance();
+
+  const connectToHardhat = useCallback(async () => {
+    try {
+      const ready = await web3SetHttpProvider(
+        (ios && HARDHAT_URL_IOS) ||
+          (android && HARDHAT_URL_ANDROID) ||
+          'http://127.0.0.1:8545'
+      );
+      logger.log('connected to hardhat', ready);
+    } catch (e) {
+      await web3SetHttpProvider(networkTypes.mainnet);
+      logger.log('error connecting to hardhat', e);
+    }
+    dispatch(explorerInit());
+    Navigation.handleAction(Routes.WALLET_SCREEN, {});
+
+    if (IS_TESTING === 'true') {
```

Comment:
i think we could remove this check, if we're connecting to hardhat we should update balances accordingly with `updateAssetOnchainBalanceIfNeeded`